lib/pbench/server/api/resources/query_apis/datasets_publish.py

```diff
@@ -0,0 +1,253 @@
+import json
+from http import HTTPStatus
+from logging import Logger
+
+from flask_restful import abort
+
+from pbench.server import PbenchServerConfig
+from pbench.server.api.resources.query_apis import (
+    API_OPERATION,
+    CONTEXT,
+    ElasticBase,
+    JSON,
+    PostprocessError,
+    Schema,
+    Parameter,
+    ParamType,
+)
+from pbench.server.database.models.tracker import Dataset, Metadata
+from pbench.server.database.models.users import User
+
+
+class DatasetsPublish(ElasticBase):
+    """
+    Change the "access" authorization of a Pbench dataset
+    """
+
+    def __init__(self, config: PbenchServerConfig, logger: Logger):
+        super().__init__(
+            config,
+            logger,
+            Schema(
+                Parameter("controller", ParamType.STRING, required=True),
+                Parameter("name", ParamType.STRING, required=True),
+                Parameter("access", ParamType.ACCESS, required=True),
```

Comment:
Well I think by asking user an `access` parameter, we are actually modifying the current access type. Here we could consider changing the name of the operation because publishing the data explicitly means we are making it "public". However, what this API operation can do is make a public dataset private as well so its two way operation not just private->public as name suggest `DatasetsPublish`. Should we consider updating the name something like `ModifyAccess` or similar?

Reply:
It's true, you can go either way. And in the long run, when we add group support, we'll have finer control over the scope of "publishing" data. I don't think that `controller/access` is a very good endpoint name, whereas `controller/publish` makes a bit more sense. I couldn't think of a better verb that's more generic. Do you have a suggestion?

Reply:
If we approach this from the point of view of "publish to whom?", then the answers might be "to the public", "to my group", "to this other group", and "to myself" (i.e., "private").  I'll grant that "publish to 'private'" isn't the best set of words, but maybe we should (or will...) change "private" rather than "publish".  (That is, I think it's good as is, for now.)

Reply:
That's true, we could "publish to self" or "publish to owner" for example, if we wanted to weasel words; and something like that may may more sense when we add groups. We'll have to re-index anyway at that point as I imagine we'll want to switch to a "group ID" rather than using a name. (Or maybe not.)

But just look at `dataset/publish` as a short-cut for "change the publication status of a dataset".